homeassistant/components/netatmo/sensor.py

```diff
@@ -112,69 +102,84 @@
 MODULE_TYPE_RAIN = "NAModule3"
 MODULE_TYPE_INDOOR = "NAModule4"
 
-
-NETATMO_DEVICE_TYPES = {
-    "WeatherStationData": "weather station",
-    "HomeCoachData": "home coach",
-}
-
 PUBLIC = "public"
 
 
-async def async_setup_entry(
-    hass: HomeAssistant, entry: ConfigEntry, async_add_entities
-):
+async def async_setup_entry(hass, entry, async_add_entities):
     """Set up the Netatmo weather and homecoach platform."""
-    auth = hass.data[DOMAIN][entry.entry_id][AUTH]
     device_registry = await hass.helpers.device_registry.async_get_registry()
+    data_handler = hass.data[DOMAIN][entry.entry_id][DATA_HANDLER]
 
-    def find_entities(data):
+    async def find_entities(data_class):
         """Find all entities."""
-        all_module_infos = data.get_module_infos()
+        await data_handler.register_data_class(data_class)
+
+        all_module_infos = {}
+
```

Comment:
```suggestion
        data = data_handler.data
```

Reply:
Can we make this a bit more readable by storing `data_handler.data` here and then using it below?